packages/multichain/src/scope/assert.ts

```diff
@@ -0,0 +1,84 @@
+import { JsonRpcError } from '@metamask/rpc-errors';
+import type { Hex } from '@metamask/utils';
+
+import {
+  isSupportedMethod,
+  isSupportedNotification,
+  isSupportedScopeString,
+} from './supported';
+import type { InternalScopeObject, InternalScopesObject } from './types';
+
+/**
+ * Asserts that a scope string and its associated scope object are supported.
+ * @param scopeString - The scope string against which to assert support.
+ * @param scopeObject - The scope object against which to assert support.
+ * @param options - An object containing the following properties:
+ * @param options.isChainIdSupported - A predicate that determines if a chainID is supported.
+ */
+export const assertScopeSupported = (
+  scopeString: string,
+  scopeObject: InternalScopeObject,
+  {
+    isChainIdSupported,
+  }: {
+    isChainIdSupported: (chainId: Hex) => boolean;
+  },
+) => {
+  const { methods, notifications } = scopeObject;
+  if (!isSupportedScopeString(scopeString, isChainIdSupported)) {
+    throw new JsonRpcError(5100, 'Requested chains are not supported');
```

Comment:
Not a blocker, but just wondering, do we intend on adding this error to `rpc-errors`? It would be ideal to have at least a constant or custom error class or something for these standardized errors

Reply:
Agreed. Would a decent intermediate (or perhaps long term) step be to create some constants in this package for the new errors specific to the Multichain API?

Reply:
Yep, that might be useful as an intermediate step, definitely

Reply:
https://github.com/MetaMask/core/pull/4784/commits/726b54e7866c58afb4166debbc2ba941377b57d2